rust/datafusion/src/sql/planner.rs

```diff
@@ -982,6 +996,171 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             ))),
         }
     }
+
+    fn sql_interval_to_literal(
+        &self,
+        value: &str,
+        leading_field: &Option<DateTimeField>,
+        leading_precision: &Option<u64>,
+        last_field: &Option<DateTimeField>,
+        fractional_seconds_precision: &Option<u64>,
+    ) -> Result<Expr> {
+        if leading_field.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with leading_field {:?}",
+                leading_field
+            )));
+        }
+
+        if leading_precision.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with leading_precision {:?}",
+                leading_precision
+            )));
+        }
+
+        if last_field.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with last_field {:?}",
+                last_field
+            )));
+        }
+
+        if fractional_seconds_precision.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with fractional_seconds_precision {:?}",
+                fractional_seconds_precision
+            )));
+        }
+
+        const SECONDS_PER_HOUR: f32 = 3_600_f32;
+        const MILLIS_PER_SECOND: f32 = 1_000_f32;
+
+        // We are storing parts as integers, it's why we need to align parts fractional
+        // INTERVAL '0.5 MONTH' = 15 days, INTERVAL '1.5 MONTH' = 1 month 15 days
```

Comment:
I found it strange that the example of`1.5 months` is not supported whereas `0.5` and `1` are supported

```
>  select (interval '0.5 month');
+-------------------------------------------------+
| IntervalDayTime("64424509440")                  |
+-------------------------------------------------+
| 0 years 0 mons 15 days 0 hours 0 mins 0.00 secs |
+-------------------------------------------------+
1 rows in set. Query took 0 seconds.
>  select (interval '1 month');
+------------------------------------------------+
| IntervalYearMonth("1")                         |
+------------------------------------------------+
| 0 years 1 mons 0 days 0 hours 0 mins 0.00 secs |
+------------------------------------------------+
1 rows in set. Query took 0 seconds.
>  select (interval '1.5 month');
NotImplemented("DF doesnt support complex intervals: \"1.1 month\"")

```

Reply:
I can understand why we might not support fractional months (as depending on the month the length of a month can not necessarily be expressed as an integral number of days or months).

But it seems inconsitent to me to say `0.5 month` is `15 days` but `1.5 month` is not `45 days`

I think the easiest thing for me to reason about is if they both error.

Reply:
I am trying to achieve similar experience as PostgreSQL in this place. Users who will use this, should know about this thing.

> NotImplemented("DF doesnt support complex intervals: \"1.1 month\"")

I think, will be great to create an RFC for Apache Arrow to introduce support for `MonthDayTime` interval type, which will help to represent any interval.

Reply:
fixed.